hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -77,16 +75,17 @@
 
     private final boolean enterprise;
 
-    private PeriodicStatistics periodicStats;
-
-    private volatile Address ownerAddress;
+    private volatile ClientConnection ownerConnection;
+    private volatile ClientConnection lastOwnerConnection;
+    private final StringBuilder stats = new StringBuilder();
 
     public Statistics(final HazelcastClientInstanceImpl clientInstance) {
         this.properties = clientInstance.getProperties();
         this.enabled = properties.getBoolean(ENABLED);
         this.client = clientInstance;
         this.enterprise = BuildInfoProvider.getBuildInfo().isEnterprise();
-        this.metricsRegistry = clientInstance.getMetricsRegistry();
+        this.context = clientInstance.getMetricsRegistry().openContext(
+                properties.getEnum(Diagnostics.METRICS_LEVEL, ProbeLevel.class));
```

Comment:
Previously client statistics were hard coded, level configuration ignored. Now client stats are equally collected which means the level is obeyed. With `MANDATORY` being the default the cache stats are only send if `INFO` is configured. It might be better to use `INFO` as default for client statistics to not change the overall behavior without explicit configuration.